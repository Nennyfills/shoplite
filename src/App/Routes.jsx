import React from 'react';
import { Switch, Route } from 'react-router-dom';

// pages
import Home from 'Pages/Home';
import NotFound from 'Pages/NotFound';

const Routes = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route  component={NotFound} />
  </Switch>
);

export default Routes;
