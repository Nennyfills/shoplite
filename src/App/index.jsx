import React from 'react';
import { ContextWrapper } from 'Utils/Store';

import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';

import Header from 'Components/Header';
import Footer from 'Components/Footer';

import Routes from './Routes';
import './app.scss';

const client = new ApolloClient({
  uri: process.env.REACT_APP_URL,
  cache: new InMemoryCache(),
});
const App = () => {
  return (
    <ApolloProvider client={client}>
      <ContextWrapper>
        <div className="app" data-test="app">
          <Header />
          <Routes />
          {/* <Footer /> */}
        </div>
      </ContextWrapper>
    </ApolloProvider>
  );
};

export default App;
