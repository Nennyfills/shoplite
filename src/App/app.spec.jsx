// import { render, screen } from '@testing-library/react';
// import App from './index';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

import React from 'react';
import { shallow } from 'enzyme';

import App from '.';

const setUp = (props = {}) => {
  const component = shallow(<App {...props} />);
  return component;
};

describe('App', () => {
  let component;
  beforeEach(() => {
    component = setUp({});
  });
  it('Should render without errors', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should render App div', () => {
    const wrapper = component.find(`[data-test='app']`);
    expect(wrapper.length).toBe(1);
  });
});
