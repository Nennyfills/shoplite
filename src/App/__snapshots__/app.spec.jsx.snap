// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render without errors 1`] = `ShallowWrapper {}`;
